civitest.module.sample

```diff
@@ -165,197 +170,174 @@ function civitest_civicrm_buildAmount(
-      'contribution.average'                   => 'Average Contribution Amount',
-      'contribution.largest'                   => 'Largest Contribution' );
+function civitest_civicrm_tokens(&$tokens) {
+  $tokens['contribution'] = array(
+    'contribution.amount' => 'Amount of contribution',
+    'contribution.date' => 'Date of contribution',
+    'contribution.total_last_year' => 'Total amount contributed last year',
+    'contribution.date_of_first' => 'Date of first contribution',
+    'contribution.largest_this_year' => 'Largest this Year Contribution Amount',
+    'contribution.date_of_largest_this_year' => 'Largest this Year Contribution Date',
+    'contribution.average' => 'Average Contribution Amount',
+    'contribution.largest' => 'Largest Contribution',
+  );
 }
 
-function civitest_civicrm_tokenValues( &$values,
+function civitest_civicrm_tokenValues(&$values,
   &$contactIDs,
   $dontCare,
   $tokens,
-  $context ) {
-  if ( is_array( $contactIDs ) ) {
-    $contactIDString = implode( ',', array_values( $contactIDs ) );
+  $context) {
+  if (is_array($contactIDs)) {
+    $contactIDString = implode(',', array_values($contactIDs));
   }
 
-  $query = "
-SELECT sum( total_amount ) as total_amount,
+  $query = "SELECT sum(total_amount) as total_amount,
        contact_id,
-       max( receive_date ) as receive_date
-FROM   civicrm_contribution
-WHERE  contact_id IN ( $contactIDString )
-AND    is_test = 0
-GROUP BY contact_id
-";
-
-  $dao = CRM_Core_DAO::executeQuery( $query );
-  while ( $dao->fetch( ) ) {
-    if ( ! array_key_exists( $dao->contact_id, $values ) ) {
-      $values[$dao->contact_id] = array( );
+       max(receive_date) as receive_date
+    FROM   civicrm_contribution
+    WHERE  contact_id IN ($contactIDString)
+    AND    is_test = 0
+    GROUP BY contact_id";
+
+  $dao = CRM_Core_DAO::executeQuery($query);
+  while ($dao->fetch()) {
+    if (!array_key_exists($dao->contact_id, $values)) {
+      $values[$dao->contact_id] = array();
     }
     $values[$dao->contact_id]['contribution.amount'] = $dao->total_amount;
-    $values[$dao->contact_id]['contribution.date'  ] = $dao->receive_date;
+    $values[$dao->contact_id]['contribution.date'] = $dao->receive_date;
   }
 }
 
 // example of token that does not depend on contactID
 // and using the new format
-function civitest_civicrm_tokenValues( &$values,
-  &$contactIDs,
-  $dontCare,
-  $tokens,
-  $context ) {
-  if ( isset( $tokens['views'] ) ) {
-    static $_cache = null;
-
-    if ( ! $_cache ) {
-      $_cache = array( );
-      foreach( $tokens['views'] as $token ) {
-        $_cache[$token] = CALL_YOUR_OWN_FUNCTION_TO_RENDER_THIS_VIEW( $token );
-      }
-    }
-
-    foreach( $contactIDs as $cid ) {
-      foreach( $tokens['views'] as $token ) {
-        $values[$cid]["views.{$token}"] = $_cache[$token];
-      }
-    }
-  }
-}
+//function civitest_civicrm_tokenValues(&$values,
```

Comment:
Why is this changed to being commented out?

(TBH, I don't think it's particularly problematic -- since the main idea of the file is to capture a bunch of code-snippets that can be copied/edited, and comments do that just as well as code... but it feels little arbitrary to comment out `hook_civicrm_tokenValues` with `//` but not anything else.)

Reply:
OK, so there's a couple more functions where the PR comments them out entirely. Same question applies to those.

Reply:
the reason is that Civilint was detecting 2 functions with the same name and i figured each had enough "good" content to warrent both of them being there so just commented one out.

Reply:
OK, that makes sense. 👍